shared/AppInsightsCore/src/JavaScriptSDK/W3cTraceParent.ts

```diff
@@ -202,3 +202,46 @@ export function findW3cTraceParent(selectIdx?: number): ITraceParent {
 
     return traceParent;
 }
+
+export interface scriptsInfo {
+    url: string;
+    crossOrigin?: string;
+    async?: boolean;
+    defer?: boolean;
+    referrePolicy?: string;
+}
+
+/**
+ * Find all script tags in the provided document and return the information about them.
+ * @param doc
+ * @returns
+ */
+export function findAllScripts(doc: any) {
+    let scripts = doc.getElementsByTagName("script");
+    let result: scriptsInfo[] = [];
+    for (let i = 0; i < scripts.length; i++) {
+        let script = scripts[i];
+        let src = script.getAttribute("src");
+        if (src) {
+            let crossOrigin = script.getAttribute("crossorigin");
+            let async = script.getAttribute("async") === "true";
```

Comment:
Depending on the browser, these may actually have no value as you only "need" to have the attribute present like

https://www.w3schools.com/tags/att_script_async.asp and https://www.w3schools.com/tags/att_script_defer.asp

So they won't necessarily have a value of "true", so you should check with `hasAttribute` and even then it's possible that the value may be false (in older browsers). So its a little complicated

Reply:
Thanks. Didn't notice that part could fail.